packages/react-native-bpk-component-icon/src/withRtlSupport.js

```diff
@@ -46,12 +46,10 @@ const getDisplayName = Component => {
   return Component.displayName || Component.name || 'Component';
 };
 
-export type Props = $Exact<BpkIconProps>;
-
 const withRtlSupport = (
-  IconComponent: ComponentType<any>,
-): ComponentType<any> => {
-  const WithRtlSupport = (props: Props) => {
+  IconComponent: typeof BpkIcon,
```

Comment:
Let's make this `ComponentType<BpkIconProps>` otherwise this will be a breaking change. Before we were accepting any component with the same prop definitions as BpkIcon and now we only accept BpkIcon.

Also I see no reason to restrict it to bpk icon only, in that case it would probably be better to just add another export with RTL support